secret/secret_test.go

```diff
@@ -120,58 +119,52 @@ func (m *MockSecrets) RecordSSLCertInfo(configVar *model.ConfigurationVariable)
 	m.Called(configVar)
 }
 
-func (m *MockSecrets) GenerateSSLCerts(secrets, updates *v1.Secret) (dirty bool) {
-	results := m.Called(secrets, updates)
-	return results.Bool(0)
+func (m *MockSecrets) GenerateSSLCerts(secrets, updates *v1.Secret) {
+	m.Called(secrets, updates)
 }
 
 func TestUpdateSecretsWhenCreatingOrUpdating(t *testing.T) {
 	t.Parallel()
 
 	t.Run("Neither Create Nor Update called", func(t *testing.T) {
 		var s MockSecretInterface
-		s.On("Create", (*v1.Secret)(nil)).Return(nil, nil)
-		s.On("Update", (*v1.Secret)(nil)).Return(nil, nil)
+		secrets := &v1.Secret{Data: map[string][]byte{}}
 
-		UpdateSecrets(&s, nil, false, false)
-		s.AssertNotCalled(t, "Create", nil)
-		s.AssertNotCalled(t, "Update", nil)
+		s.On("Create", secrets).Return(nil, nil)
+		s.On("Update", secrets).Return(nil, nil)
+
+		UpdateSecrets(&s, secrets, false)
+		s.AssertNotCalled(t, "Create", secrets)
+		s.AssertNotCalled(t, "Update", secrets)
```

Comment:
Assert that the secrets are not dirty? That it's empty?

Reply:
I don't know; `UpdateSecrets` doesn't do anything to the `secrets` in-memory structure except for resetting the dirty flag.

There are endless things that should *not* happen during the call; why do you want to test for this particular thing? Same for the other calls to `UpdateSecrets` (except that I did mark the secrets as `dirty` and now check that they are clean when the function returns).

Reply:
Mostly, I just wanted something to ensure that we don't leak the dirtiness flag into the actual secret store. I'm happy however you desire to check that.